operate/webapp/src/main/java/io/camunda/operate/webapp/zeebe/operation/AbstractOperationHandler.java

```diff
@@ -70,8 +69,8 @@ public void handle(final OperationEntity operation) {
 
   // Needed for tests
   @Override
-  public void setCamundaClient(final CamundaClient camundaClient) {
-    this.camundaClient = camundaClient;
+  public void setOperateAdapter(final OperateServicesAdapter operateAdapter) {
+    operationServicesAdapter = operateAdapter;
   }
 
   private boolean isExceptionRetriable(final Exception ex) {
```

Comment:
🔧 It checks a specific gRPC exception (to decide if the operation can be retried), which won't be thrown when using the `ServicesBasedAdapter`. Basically, the `ServicesBasedAdapter` should handle the retriable exception and potentially throw a proper exception that can be handled here (independent of whether it is a `ServicesBased` or `ClientBased` adapter).

Reply:
👍 I'll move this method to the specific Adapter implementations. The existing code will exist in the `ClientBasedAdapter`, and I'll implement the method for the specific Service-related exceptions.

That way, we keep the specific exceptions code separate from the "general" `AbstractOperationHandler` code.